animatplot/xanim.py

```diff
@@ -0,0 +1,61 @@
+import animatplot as amp
+import xarray as xr
+
+
+def animated_plot(da, anim_over='__guess__', x='x', y='y', plot_type='imshow', fps=10):
```

Comment:
The core of xarray is still a numpy array. I sort of think that ```anim_over``` should be ```t_axis```. That said, the xarray docs keep using the term "coods", so maybe ```t_coord```? I think I prefer ```t_axis``` for api consistency. OTOH, t_axis is usually an integer.... 

Either way, the default value should be ```None```, and you can check
```
if anim_over is None:
...
```

Also, could you add the numpydoc style parameter to the docstring? Looks like:
```
Parameters
------------
parameter_name : type
    description
…
Returns
--------
type1
    description
type2
   description
```
You can read about numpydoc [here](https://numpydoc.readthedocs.io/en/latest/format.html#sections)

Reply:
> I sort of think that anim_over should be t_axis. 

That would defeat a lot of the purpose of using xarray. The point of xarray is that the dimensions of the data are labelled in a human-readable manner, and you can interact with the data without specifying seemingly-arbitrary numbers (as in `axis=0`).

> the xarray docs keep using the term "coords"

The xarray docs use the term `coord` because that is one of the fundamental objects in xarray, and it's not the same thing as a dimension (which is not the same thing as an axis!).

To be honest when I first started using xarray I was like "is all this extra jargon really necessary?", but having used it for a while it makes way more sense.

> the default value should be None

Yep you're right.

> I think I prefer t_axis for api consistency

I didn't use the word "axis" because that implies an integer numpy axis, which is what xarray allows you move past. I didn't use the letter "t" because `t_axis` only represents time in the sense that a gif evolves over real time. In general then `t_axis` could refer to something which isn't time at all. For example, I could have a DataArray of temperature T(x,y,z,t) in the atmosphere for each day in a month. If I want to plot a gif of the variation of temperature with height ona  particular day then it would be much clearer to be able to write `animate(da, anim_over='z')`  or even `animate(da, anim_over='height')` then to write `animate(da, t_axis='z')` (or worse, `animate(da, t_axis=3)`. I chose "`anim_over`" because it's literally the dimension of the data over which the gif is to be animated. What do you think?

> numpydoc

Yep, I am more than happy to conform to animatplot's commenting and documenting style (and write some unit tests), I just wanted to get the general idea of the layout worked out first.

Reply:
Ok, ```anim_over``` it is.